src/book/02-system/04-hardware/04-dynamixel-protocol.mdx

```diff
@@ -0,0 +1,388 @@
+
+## Duplexity
+
+Firstly, one must understand the concept of duplexity since the half duplexity of the Dynamixel protocol is not always intuitive for beginners. There are three kinds of digital communication, namely simplex, full duplex, and half duplex.
+
+Simplexity is when data goes in only one direction, from Device-A to Device-B, where only the former transmits and only the latter receives. There is only one channel, and there may often be only one electrical signal. An analogy would be Alice giving instructions to Bob through a loud-speaker very far away.
+
+![In simplexity, data always flows in only one direction.](./images/dynamixel_protocol/simplexity.drawio.png 'In simplexity, data always flows in only one direction.')
+
+Full duplexity is when data can go in both directions at the same time, between Device-A and Device-B. There are two channels, one for each direction such that one could think of each channel as simplex. There is no clash when both Device-A and Device-B send data at the same time since the two channels are separate. An analogy would be Alice and Bob writing letters or texting.
+
+![In full duplexity, data can flow in both directions at the same time.](./images/dynamixel_protocol/full_duplexity.drawio.png 'In full duplexity, data can flow in both directions at the same time.')
+
+A pure UART will have two signals, RX and TX, and if it were connected to another UART such that TX->RX both ways, then this would be an example of full duplexity.
+
+Half duplexity is when data can go in both directions but not at the same time. There seem to be two channels but are really combined into one. An analogy would be Alice and Bob talking to each other by walkie-talkies.
+
+![In half duplexity, data can flow in both directions but not at the same time.](./images/dynamixel_protocol/half_duplexity.drawio.png 'In half duplexity, data can flow in both directions but not at the same time.')
+
+<Alert type="warn">
+
+The Dynamixel protocol is half duplex; therefore, it is important to understand half duplexity and the needed flow-control and logic behind it.
+
+</Alert>
+
+We will see soon that the reason why the protocol is half duplex is the fewer number of wires - especially for RS485 - and the fact that servos only respond to instructions.
+
+Often, a Dynamixel device's microcontroller has a UART to communicate on the bus. This UART as said before has two pins, RX and TX. Therefore, in theory, it could be used for full duplexity. However, RX and TX get switched between by a transceiver which is often another chip on the board. This transceiver as the name suggests can switch between receiving and transmitting on the same bus-line. Therefore, this transceiver has another input pin to set the direction, sometimes called the TX-enable, the direction, etc.
```

Comment:
```suggestion
Often, a Dynamixel device's microcontroller has a UART to communicate on the bus. This UART has two pins, RX and TX. Therefore, in theory, it could be used for full duplexity. However, RX and TX get switched between by a transceiver, which is often another chip on the board. The transceiver can switch between receiving and transmitting on the same bus-line. This transceiver has another input pin to set the direction, sometimes called the TX-enable, the direction, etc.
```
I was at first unsure why you said 'often' but then I thought to the case where a dynamixel device is a custom device, like our FSRs. Maybe it'd be useful to mention this.